doc/how_to/custom_components/esm/custom_layout.md

```diff
@@ -230,153 +225,215 @@ radio_button_group.servable()
+
+      let splits = [];
+
+      objects.forEach((object, index) => {
+        const split = document.createElement('div');
+        splits.push(split)
+
+        splitDiv.appendChild(split);
+        split.appendChild(object);
+      })
+
+      Split(splits, {direction: 'vertical'})
+
+      return splitDiv
+    }"""
+
+    _stylesheets = [CSS]
+
+
+pn.extension("codeeditor")
+
+grid_js = GridJS(
+    pn.widgets.CodeEditor(
+        value="I love beatboxing\n" * 10, theme="monokai", sizing_mode="stretch_both"
+    ),
+    pn.panel(
+        "https://upload.wikimedia.org/wikipedia/commons/d/d3/Beatboxset1_pepouni.ogg",
+        sizing_mode="stretch_width",
+        height=100,
+    ),
+    pn.widgets.CodeEditor(
+        value="Yes, I do!\n" * 10, theme="monokai", sizing_mode="stretch_both"
+    ),
     styles={"border": "2px solid lightgray"},
+    height=800,
+    width=500,
+    sizing_mode="fixed",
 ).servable()
 ```
 
-You must list `NamedListLike, Layoutable, Viewer` in exactly that order when you define the class! Other combinations might not work.
+You must list `NamedListLike, JSComponent` in exactly that order when you define the class! Reversing the order to `JSComponent, NamedListLike` will not work.
 
 :::
 
-:::{tab-item} `JSComponent`
+:::{tab-item} `ReactComponent`
 
 ```{pyodide}
 import panel as pn
 import param
-from panel.custom import JSComponent
+from panel.custom import ReactComponent
 from panel.layout.base import NamedListLike
 
-pn.extension()
+CSS = """
+.gutter {
+    background-color: #eee;
+    background-repeat: no-repeat;
+    background-position: 50%;
+}
+.gutter.gutter-vertical {
+    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAFAQMAAABo7865AAAABlBMVEVHcEzMzMzyAv2sAAAAAXRSTlMAQObYZgAAABBJREFUeF5jOAMEEAIEEFwAn3kMwcB6I2AAAAAASUVORK5CYII=');
+
+
+ cursor: row-resize;
+}
+"""
 
 
-class ListLikeLayout(NamedListLike, JSComponent):
+class GridReact(NamedListLike, ReactComponent):
     objects = param.List()
 
     _esm = """
-    export function render({ model }) {
-      const div = document.createElement('div')
-      let objects = model.get_child("objects")
-
-      objects.forEach((object, index) => {
-        div.appendChild(object);
-
-        // If it's not the last object, add a divider
-        if (index < objects.length - 1) {
-            const divider = document.createElement("div");
-            divider.className = "divider";
-            div.appendChild(divider);
-        }
-        });
-      return div
+    import Split from 'https://esm.sh/react-split@2.0.14'
+
+    export function render({ model}) {
+      const objects = model.get_child("objects")
+      console.log(objects)
+      const calculatedHeight = `calc( 100% - ${(objects.length - 1) * 10}px )`;
+
+      return (
+        <Split
+            className="split"
+            direction="vertical"
+            style={{ height: model.height }}
```

Comment:
I would have like to just put in `height: 100%` but it won't work because the `el` does not stretch to the height and width of its parent. I'm not sure what will happen if I dynamically change the `GridReact.height` value?

Reply:
Works fine now.